e2e/src/test/java/org/jdbi/v3/e2e/TestIssue2623.java

```diff
@@ -0,0 +1,120 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.e2e;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Objects;
+
+import com.google.common.collect.ImmutableMultimap;
+import com.google.common.collect.Multimap;
+import org.jdbi.v3.core.generic.GenericType;
+import org.jdbi.v3.core.mapper.RowMapper;
+import org.jdbi.v3.core.statement.StatementContext;
+import org.jdbi.v3.guava.GuavaPlugin;
+import org.jdbi.v3.sqlobject.SqlObjectPlugin;
+import org.jdbi.v3.sqlobject.config.KeyColumn;
+import org.jdbi.v3.sqlobject.config.RegisterConstructorMapper;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.jdbi.v3.testing.junit5.JdbiExtension;
+import org.jdbi.v3.testing.junit5.internal.TestingInitializers;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import static org.assertj.guava.api.Assertions.assertThat;
+
+public class TestIssue2623 {
+
+    @RegisterExtension
+    public JdbiExtension h2Extension = JdbiExtension.h2()
+        .withInitializer(TestingInitializers.usersWithData())
+        .withPlugins(new SqlObjectPlugin(), new GuavaPlugin());
+
+    @BeforeEach
+    public void setUp() {
+        h2Extension.getJdbi().registerRowMapper(new GenericType<>() {}, new UserIntegerMapper());
+    }
+
+    @Test
+    public void testParameterizeTypedMapper() {
+        var jdbi = h2Extension.getJdbi();
+
+        Multimap<Integer, User<Integer>> result = jdbi.withExtension(Dao.class, dao -> dao.getUsers(1));
+        assertThat(result).isNotNull().containsAllEntriesOf(ImmutableMultimap.of(1, new User<>("Alice", 1)));
+    }
+
+    @Test
+    public void testConcreteTypedMapper() {
+        var jdbi = h2Extension.getJdbi();
+
+        Multimap<Integer, IntUser> result = jdbi.withExtension(Dao.class, dao -> dao.getIntUsers(1));
+        assertThat(result).isNotNull().containsAllEntriesOf(ImmutableMultimap.of(1, new IntUser("Alice", 1)));
+    }
+
+    public interface Dao {
+        @SqlQuery("select name, id from users where id = :id")
+        @KeyColumn("id")
+        Multimap<Integer, User<Integer>> getUsers(int id);
+
+        @SqlQuery("select name, id from users where id = :id")
+        @KeyColumn("id")
+        @RegisterConstructorMapper(IntUser.class)
+        Multimap<Integer, IntUser> getIntUsers(int id);
+    }
+
+    public record User<T>(String name, T id) {}
+
+    public static class UserIntegerMapper implements RowMapper<User<Integer>> {
```

Comment:
This might really neatly be replaced with a `ConstructorMapper`, unless showing this mapper is part of the point of the example.

Reply:
showing the mapper is the point of the example.